plasmapy/diagnostics/proton_radiography.py

```diff
@@ -0,0 +1,1203 @@
+"""
+Routines for the analysis of proton radiographs. These routines can be broadly
+classified as either creating synthetic radiographs from prescribed fields or
+methods of 'inverting' experimentally created radiographs to reconstruct the
+original fields (under some set of assumptions).
+"""
+
+__all__ = [
+    "TestField",
+    "ElectrostaticGaussianSphere",
+    "AxiallyMagnetizedCylinder",
+    "ElectrostaticPlanarShock",
+    "test_fields",
+    "SyntheticProtonRadiograph",
+]
+
+import astropy.constants as const
+import astropy.units as u
+import numpy as np
+import scipy.interpolate as interp
+import time
+import warnings
+
+from scipy.special import erf as erf
+
+
+class TestField:
+    """
+    Base class for example fields for testing SyntheticProtonRadiograph
+    functions.
+    """
+
+    def __init__(self, grid, emax=0 * u.V / u.m, bmax=0 * u.T, regular_grid=None):
+        self.grid = grid
```

Comment:
Add doctrings.  You don't need to include a 2nd description of the class, just the parameter descriptions.